examples/counter/cli/src/main.rs

```diff
@@ -42,64 +49,116 @@ struct Args {
 
 #[async_std::main]
 async fn main() -> Result<()> {
-    let mut queue: VecDeque<CoreMessage> = VecDeque::new();
+    let (tx, rx) = crossbeam_channel::unbounded::<CoreMessage>();
 
     let cmd = Args::parse().cmd;
-    queue.push_back(cmd.into());
-
-    while !queue.is_empty() {
-        let msg = queue.pop_front();
-
-        let reqs = match msg {
-            Some(CoreMessage::Message(m)) => shared::message(&bcs::to_bytes(&m)?),
-            Some(CoreMessage::Response(uuid, output)) => shared::response(
-                &uuid,
-                &match output {
-                    Outcome::Http(x) => bcs::to_bytes(&x)?,
-                },
-            ),
-            _ => vec![],
-        };
-        let reqs: Vec<Request<Effect>> = bcs::from_bytes(&reqs)?;
 
-        for Request { uuid, effect } in reqs {
-            match effect {
-                Effect::Render(_) => (),
-                Effect::Http(HttpRequest { method, url }) => {
-                    let method = Method::from_str(&method).expect("unknown http method");
-                    let url = Url::parse(&url)?;
-                    let response = http(method, &url).await?;
+    let watching = matches!(cmd, Command::Watch);
+    if watching {
+        println!("^C to exit");
+    }
 
-                    queue.push_back(CoreMessage::Response(uuid, Outcome::Http(response)));
+    tx.send(cmd.into()).unwrap();
+
+    let handle = async_task_group::group(|group| async move {
+        'outer: while let Ok(msg) = rx.recv() {
+            let reqs = match msg {
+                CoreMessage::Message(m) => shared::message(&to_bytes(&m).unwrap()),
+                CoreMessage::Response(uuid, output) => shared::response(
+                    &uuid,
+                    &match output {
+                        Outcome::Http(x) => to_bytes(&x).unwrap(),
+                        Outcome::Sse(x) => to_bytes(&x).unwrap(),
+                    },
+                ),
+            };
+            let reqs: Vec<Request<Effect>> = from_bytes(&reqs).unwrap();
+
+            for Request { uuid, effect } in reqs {
+                match effect {
+                    Effect::Render(_) => {
+                        let view = from_bytes::<ViewModel>(&shared::view())?;
+                        if !view.text.contains("pending") {
```

Comment:
This is not ideal. If we were able to track sender refs instead, we could terminate after the channel is dropped.